pages/_app.tsx

```diff
@@ -1,7 +1,13 @@
 import { AppProps } from 'next/app'
+import { useRouter } from 'next/router'
+import { useEffect } from 'react'
 
 import '../lib/global.css'
 
 export default function MyApp({ Component, pageProps }: AppProps) {
+  const router = useRouter()
+  useEffect(() => {
+    document.cookie = 'ref=' + router.query.ref
```

Comment:
have to make sure it's unique, otherwise other things like browser extensions could clobber it
```suggestion
    document.cookie = 'fc-referral-ref=' + router.query.ref
```

secondly, you need to explicitly set the `domain` to `flightcontrol.dev`. Otherwise it will default to `www.flightcontrol.dev` and won't be accessible from `app.flightcontrol.dev`